src/patterns.jl

```diff
@@ -64,14 +64,17 @@ julia> filter(in(Pattern(:__, :load, :_)), items)
             end
         end
 
-        return new(segments[mask])
+        vals = segments[mask]
+        return new{typeof(vals)}(vals)
     end
 end
 
-Pattern(segments::Symbol...) = Pattern(segments)
-Base.convert(::Type{Pattern}, x::Tuple{Vararg{Symbol}}) = Pattern(x)
+Pattern(segments...) = Pattern(segments)
+Base.convert(::Type{Pattern}, x::Tuple) = Pattern(x)
 
-function Base.in(item::Tuple{Vararg{Symbol}}, pattern::Pattern)
+Base.show(io::IO, pattern::Pattern) = print(io, "Pattern($(pattern.segments))")
```

Comment:
`show` method was added cause the now parameterized `Pattern` type was a bit verbose. Especially, since you can just look at the values most of the time.